src/electionguard/decryption_mediator.py

```diff
@@ -132,189 +152,275 @@ def announce(self, guardian: Guardian) -> Optional[TallyDecryptionShare]:
         # before announcing a guardian is available for decryption.
         for guardian_id, public_key in missing_guardians.items():
             if guardian_id in self._missing_guardians.keys():
-                if self._missing_guardians.get(guardian_id) != public_key:
+                if self._missing_guardians[guardian_id] != public_key:
                     log_warning(
                         (
                             f"announce guardian: {guardian.object_id} "
                             f"expected public key mismatch for missing {guardian_id}"
                         )
                     )
-                    return None
+                    return False
             else:
-                self._missing_guardians.set(guardian_id, missing_guardians[guardian_id])
+                self._missing_guardians[guardian_id] = missing_guardians[guardian_id]
+        return True
+
+    # pylint: disable=too-many-return-statements
+    def get_plaintext_tally(
+        self, decrypt: AuxiliaryDecrypt = rsa_decrypt
+    ) -> Optional[PlaintextTally]:
+        """
+        Get the plaintext tally for the election by composing each Guardian's
+        decrypted representation of each selection into a decrypted representation
+
+        :return: a `PlaintextTally` or `None`
+        """
+
+        # Make sure a Quorum of Guardians have announced
+        if len(self._available_guardians) < self._encryption.quorum:
+            log_warning(
+                "cannot get plaintext tally with less than quorum available guardians"
+            )
+            return None
+
+        # If all Guardians are present decrypt the tally
+        if len(self._available_guardians) == self._encryption.number_of_guardians:
+            return decrypt_tally(
+                self._ciphertext_tally,
+                self._tally_shares,
+                self._encryption.crypto_extended_base_hash,
+            )
+
+        # If guardians are missing, compensate then decrypt
+        self._compute_missing_shares_for_tally(decrypt)
+
+        if len(self._tally_shares) != self._encryption.number_of_guardians:
+            log_warning("get plaintext tally failed with share length mismatch")
+            return None
+
+        return decrypt_tally(
+            self._ciphertext_tally,
+            self._tally_shares,
+            self._encryption.crypto_extended_base_hash,
+        )
+
+    def _compute_missing_shares_for_tally(
```

Comment:
also suggest adding a summary comment to this method so a developer can get intellisense without having to navigate